functions/src/index.ts

```diff
@@ -65,28 +109,67 @@ exports.thingspeakToFirestore = functions
       );
 
       const resolvedPath = readingsSubcollection(knownSensor.id);
-
-      // Only add data if not already present in database.
-      // This happens if a sensor is down, so only old data is returned.
       const readingsRef = firestore.collection(resolvedPath);
+
+      // If data is not already present in the database add
+      // it to the historical readings
+      let firestoreSafeReading = defaultPm25BufferElement;
       if (
         (
           await readingsRef
             .where('timestamp', '==', Timestamp.fromDate(reading.timestamp))
             .get()
         ).empty
       ) {
-        const firestoreSafeReading = {
+        // Update reading with values
+        firestoreSafeReading = {
           timestamp: Timestamp.fromDate(reading.timestamp),
           channelAPm25: reading.channelAPm25,
           channelBPm25: reading.channelBPm25,
           humidity: reading.humidity,
           latitude: reading.latitude,
           longitude: reading.longitude,
         };
+        // Add to historical readings
         await readingsRef.add(firestoreSafeReading);
       }
 
+      // Add readings to the pm25 buffer
+      const docRef = firestore.collection('sensors').doc(knownSensor.id);
```

Comment:
Why do we do another read here? Could we do this inside of the existing if statement using the same information from the data we're about to write to Firestore?

Reply:
After reading more code, is this supposed to be a different collection?

Reply:
Adam's original code writes to a new document in the readings sub collection. This code writes to the fields in the sensor doc. I'm not really sure what this comment is getting at.